lib/exception_notifier/email_notifier.rb

```diff
@@ -0,0 +1,242 @@
+require 'action_mailer'
+require 'action_dispatch'
+require 'pp'
+
+module ExceptionNotifier
+  class EmailNotifier < Struct.new(:sender_address, :exception_recipients,
+    :email_prefix, :email_format, :sections, :background_sections,
+    :verbose_subject, :normalize_subject, :delivery_method, :mailer_settings,
+    :email_headers, :mailer_parent, :template_path)
+
+    module Mailer
+      class MissingController
+        def method_missing(*args, &block)
+        end
+      end
+
+      def self.extended(base)
+        base.class_eval do
+          # Append application view path to the ExceptionNotifier lookup context.
+          self.append_view_path "#{File.dirname(__FILE__)}/views"
+
+          @@whitelisted_env_vars = [
+            'action_dispatch.request.parameters',
+            'action_dispatch.request.path_parameters',
+            'action_dispatch.request.query_parameters',
+            'action_dispatch.request.request_parameters',
+            'BUNDLE_BIN_PATH',
+            'BUNDLE_GEMFILE',
+            'CONTENT_LENGTH',
+            'CONTENT_TYPE',
+            'DOCUMENT_ROOT',
+            'GEM_HOME',
+            'HOME',
+            /HTTP_/,
+            'ORIGINAL_FULLPATH',
+            'PASSENGER_APP_TYPE',
+            'PASSENGER_ENV',
+            'PASSENGER_RUBY',
+            'PASSENGER_SPAWN_METHOD',
+            'PASSENGER_USER',
+            'PATH',
+            'PATH_INFO',
+            'PWD',
+            'RAILS_ENV',
+            'REMOTE_ADDR',
+            'REMOTE_PORT',
+            'REQUEST_METHOD',
+            'REQUEST_URI',
+            'RUBYOPT',
+            'SERVER_ADDR',
+            'SERVER_NAME',
+            'SERVER_PORT',
+            'SERVER_PROTOCOL',
+            'SERVER_SOFTWARE',
+            'TMPDIR',
+            'USER',
+          ]
+
+
+          #NB: This doesn't seem to correctly create instance attr from
+          #    within base#class_eval:
+          #cattr_accessor :whitelisted_env_vars
```

Comment:
I _think_ this is because `cattr_accessor` also uses `#class_eval` so you end-up running one nested inside the other, which means the `cattr_accessor` version is working on the `Class` object, therefore doesn't create the class-method in the right place.
Be good if someone could comment on this (@willbryant ?)